### PR TITLE
Allow warnings to be hidden

### PR DESCRIPTION
**Pull request by jennyrobertsonbbc**
## Summary

Fix issue https://github.com/bbc/bbc-a11y/issues/303. Allow warnings to be hidden in the same way that errors can be hidden.

## Details

I made the `addWarning` method work in the same way as `addError`. I also added a test.

I added `Warnings hidden` to the output:
image

## Motivation and Context

Fix issue https://github.com/bbc/bbc-a11y/issues/303

## How Has This Been Tested?

I added a test for the new code which is very similar to the test for hiding errors.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I've added tests for my code
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: features/hiding_warnings.feature [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        hide: "First heading was not a main heading"
+      })
+      """
+    When I run `bbc-a11y`
+    Then it should pass with:
+      """
+      ✓ http://localhost:54321/one_warning.html
+
+      1 page checked, 0 errors found, 0 warnings, 1 warning hidden, 0 standards skipped
+      """
```
Comment (andymsuk) [inline features/hiding_warnings.feature]:
```suggestion
      """

```

File: features/support/web_server/one_warning.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en-gb">
+  <head>
+    <title>Blank</title>
+  </head>
+  <body>
+    <div id='main' role='main'>
```
Comment (andymsuk) [inline features/support/web_server/one_warning.html]:
```suggestion
    <div id="main" role="main">
```

**Comment by andymsuk** [comment]
Looks good to me from a code point of view.

@EmmaJP / @micmath / @JamieKnightBBC - any thoughts/comments from you?

**Comment by JamieKnightBBC** [comment]
Looks good to me. A handy feature which aligns things nicely.

From a code perspective, and this is more a style thing than anything, there’s a line:

````if (this.hide) {````

I’m not sure what exactly that value is and the condition being checked for. Should it be undefined? True, false, 0 etc.

I’d consider making this more explicit to aid code readability and robustness.

**Comment by jennyrobertsonbbc** [comment]
> Looks good to me. A handy feature which aligns things nicely.
> 
> From a code perspective, and this is more a style thing than anything, there’s a line:
> 
> `if (this.hide) {`
> 
> I’m not sure what exactly that value is and the condition being checked for. Should it be undefined? True, false, 0 etc.
> 
> I’d consider making this more explicit to aid code readability and robustness.

Thanks for your feedback :). I copied the existing function (`addError`) when I made `addWarning` and thats how it was in there. I am happy to try to rename the variable in both functions to make it clearer.

**Comment by andymsuk** [comment]
@JamieKnightBBC Yeah, `this.hide` is used elsewhere. I think it's fine - it's just checking for the presence of the `hide` option (checking for a truthy value) and a few lines down, it looks at the value to determine what warnings to hide.

**Comment by andymsuk** [comment]
@JamieKnightBBC - happy to approve?

**Comment by ermi-ltd** [comment]
Heyo,

Good with me. Was a small style question. Good to have it match what everything else is doing :)

J&L

**Comment by ermi-ltd** [comment]
Gah, wrong account again. My fault for replying from my phone. _blush_

**Pull request merged**